cidr-filtering-cni

```diff
@@ -0,0 +1,252 @@
+    local direction="$3"
+    local match_iface="$4"
+    
+    local match_addr="saddr"
+    if [[ "$direction" == "egress" ]]; then
+        local match_addr="daddr"
+    fi
+    
+    _create_subnet_rule() {
+        local subnet=$1
+        if [[ "$(get_object "${subnet}" ".subnet.except")" != "null" ]]; then
+            for_json_array "${subnet}" ".subnet.except" _drop_ip
+        fi
+
+        if [[ "$(get_object "${subnet}" ".subnet.cidr")" == "null" ]]; then
+            exitWithError "cidr must be specified."
+        fi
+        _accept_cidr "${subnet}"
+    }
+
+    _drop_ip() {
+        local ip="$1"
+        ip netns exec "$CNI_CONTAINERID" $(nft_add_rule ${table_type} ${NFT_TABLE} m-${policy_id}-${direction}-subnets ${match_iface} $CNI_IFNAME $(get_ip_version "${ip}") ${match_addr} ${ip} counter drop)
+    }
+
+    _accept_cidr() {
+        local subnet="$1"
+        local cidr=$(echo "${subnet}" | jq -r ".subnet.cidr")
+        ip netns exec "$CNI_CONTAINERID" $(nft_add_rule ${table_type} ${NFT_TABLE} m-${policy_id}-${direction}-subnets ${match_iface} $CNI_IFNAME $(get_ip_version "${cidr}") ${match_addr} ${cidr} counter meta mark set mark or 0x20000)
+    }
+
+    _accept_port() {
+        local ports="$1"
+        local port=$(echo "$ports" | jq -r ".port")
+        local protocol=$(echo "$ports" | jq -r ".protocol")
+
+        if [[ "$port" == "null" || "$protocol" == "null" || "$port" == "" || "$protocol" == "" ]]; then
+            exitWithError "Port and protocol must be specified."
+        else
+            ip netns exec "$CNI_CONTAINERID" $(nft_add_rule ${table_type} ${NFT_TABLE} m-${policy_id}-${direction}-ports ${match_iface} $CNI_IFNAME ${protocol,,} dport ${port} counter meta mark set meta mark "|" 0x00010000)
+        fi
+    }
+
+    # handle ip block
+    if [[ $(echo "$stdin" | jq -r ".${direction}.subnets") != "null" ]]; then
+        for_json_array "$stdin" ".${direction}.subnets" _create_subnet_rule
+    fi
+
+    # handle ports
+    if [[ $(echo "$stdin" | jq -r ".${direction}.ports") != "null" ]]; then
+        for_json_array "$stdin" ".${direction}.ports" _accept_port
+    fi
+}
+
+exitWithError() {
+    echo "{"cniVersion": "$(jq -r ".cniVersion" <<< "$stdin")","msg":"$1","code":1,"details":"$2"}" >&3
+    exit 1
+}
+
+exitWithSuccess() {
+    prev_result=$(get_object "${stdin}" ".prevResult")
+    if [[ "$prev_result" == "null" ]]; then
+        echo "{"cniVersion": "$(jq -r ".cniVersion" <<< "$stdin")"}" >&3
+    else
+        echo "$prev_result" >&3
+    fi
+    exit 0
+}
+
+create_chains_for_filtering() {
+    local policy_id="$1"
+    local table_type="$2"
+    local direction="$3"
+
+    local match_iface="iifname"
+    if [[ "$direction" == "egress" ]]; then
+        local match_iface="oifname"
+    fi
+
+    ip netns exec "$CNI_CONTAINERID" $(create_chain "${table_type}" "${NFT_TABLE}" m-"${policy_id}"-"${direction}") || exitWithError "Failed to add chain m-${policy_id}-${direction}"
```

Comment:
please make sure that the fact you have `|| exitWithError` doesn't make this function disregard -e all over
there is a gotcha in bash about it IIRC

Reply:
I tested the behaviour and it seems to work as desired. Don't you mix it up with a pipe?

Because for example `false | echo hello ` succeeds with $? == 0. That is why I  `set -o pipefail` to carry failures forward when in piped commands

Reply:
not mixing it with a pipe, if a function has `|| return true` on a command, it will disregard the set -e
(i might not remember the exact combination, but it wasn't a pipe)
anyhow if it works good then all good